products/swift-build-tool/swift-build-tool.cpp

```diff
@@ -44,6 +44,11 @@ class BasicBuildSystemFrontendDelegate : public BuildSystemFrontendDelegate {
 
     return nullptr;
   }
+
+  virtual void cycleDetected(const std::vector<core::Rule*>& items) override {
+    // FIXME.
+    assert(0 && "unexpected cycle!");
+  }
```

Comment:
What if we made the frontend just provide a base implementation of this, and clients can choose whether to call it or not?

We don't want to break cycle reporting for SwiftPM, so we have to do something here.